src/App.tsx

```diff
@@ -1,41 +1,110 @@
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { getPosts, getUserPosts } from './api/posts';
+import { getUsers } from './api/users';
+import { Loader } from './components/Loader';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectUserId, setSelectUserId] = useState(0);
+  const [selectPostId, setSelectPostId] = useState(0);
+  const [isLoader, setLoader] = useState(false);
+
+  const loadPosts = async () => {
+    const postsFromServer = await getPosts();
+
+    setPosts(postsFromServer);
+  };
+
+  const loadPostsByUserId = async () => {
+    const postsFromServer = await getUserPosts(selectUserId);
+
+    setPosts(postsFromServer);
+  };
+
+  const loadUsers = async () => {
+    const usersFromServer = await getUsers();
+
+    setUsers(usersFromServer);
+  };
+
+  const loadAllData = async () => {
+    setLoader(true);
+    await loadPosts();
+    await loadUsers();
+    setLoader(false);
+  };
+
+  useEffect(() => {
+    loadAllData();
+  }, []);
+
+  useEffect(() => {
+    if (selectUserId !== 0) {
+      loadPostsByUserId();
+    } else {
+      loadPosts();
+    }
+  }, [selectUserId]);
+
+  const hendlerSelect = useCallback((event: React.ChangeEvent<HTMLSelectElement>) => {
```

Comment:
the same in all similar namings

```suggestion
  const handleSelect = useCallback((event: React.ChangeEvent<HTMLSelectElement>) => {
```